utils/models_setup.py

```diff
@@ -30,7 +30,9 @@ def get_device(use_gpu=True, rank_per_model=1):
         raise ValueError("Exactly 1 rank per device currently supported")
 
     print("Using GPU")
-    device_name = "cuda:" + str(world_rank)
+    ## We try not to specify GPU IDs to be more flexible
+    #device_name = "cuda:" + str(world_rank)
+    device_name = "cuda"
```

Comment:
But won't this break any cluster that doesn't have a resource set concept? Can we just use 1 resource set instead?

Reply:
I changed to use "OMPI_COMM_WORLD_LOCAL_RANK" (Summit) or "SLURM_LOCALID" (CADES) to set device id.